linked-lists/c-or-cpp/singly.cpp

```diff
@@ -34,6 +34,27 @@ class SinglyLinkedList {
             // change head point to new node
             this->head = temp;
         }
+    
+        void insertAtEnd(T data){
+
+            Node<T> *newNode = new Node<T>(data);
+
+            // if head is not pointing to anything then simply point to new node
+            if(this->head == nullptr) {
+                this->head = newNode;
+                return;
+            }
+
+            // if list is not empty, then traverse to the last node using `temp`, then put the address of the new node in the temp->next
+            // and finally, make the next of newNode = NULL
+            Node<T> *temp = new Node<T>();
```

Comment:
This line will cause an error, as there is not matching overload of this function available in template class.  

you can check [this](https://ide.geeksforgeeks.org/ZsPhitAhUD) for more details about the same

Reply:
Sorry, I’ve added an empty constructor, but i will add it again.

Reply:
It is now working good